harmony_service_lib/exceptions.py

```diff
@@ -33,3 +36,10 @@ class ServerException(HarmonyException):
 
     def __init__(self, message=None):
         super().__init__(message, 'Server')
+
+
+class NoDataException(HarmonyException):
+    """Class for throwing an exception indicating service found no data to process """
+
+    def __init__(self, message=None):
```

Comment:
I'm assuming this message is being written to the warnings file and is the message that will be displayed in the warnings field for the job status. If that's the case we should provide a default message of "No data found to process" or similar, but allow a service to provide their own message to override it if they want.

Reply:
Updated default message.